plugins/nextcloud-rule-exclusions-before.conf

```
@@ -739,15 +792,38 @@ SecRule REQUEST_FILENAME "@endsWith /login" \
             ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:pass1-text,\
             ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:pass2"
 
-# Logout token
-SecRule REQUEST_FILENAME "@endsWith /logout" \
-    "id:9508420,\
+# Password reset used in newer version of Nextcloud
+SecRule REQUEST_FILENAME "@rx /lostpassword/set/[^/]+/[^/]+$" \
+    "id:9508411,\
     phase:1,\
     pass,\
     t:none,\
     nolog,\
     ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
-    ctl:ruleRemoveTargetById=941120;ARGS:requesttoken"
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.password"
+
+# requesttoken is used when logging in or when authenticating via a public share
+# Matches:
+# /login
+# /s/share-id/authenticate/showShare
+# /login/v2/grant
+# /logout
+SecRule REQUEST_FILENAME "@rx /(?:login(?:/v[0-9\.]+/grant)?|s/[^/]+/authenticate/showShare|logout)$" \
```

Should also match `/login`, to support original rule 9508400.

---

it should match /login, does it not? 

or is the regex not clear enough, should I change to this?
```suggestion
SecRule REQUEST_FILENAME "@rx /(?:login|login/v[0-9\.]+/grant|s/[^/]+/authenticate/showShare|logout)$" \
```

---

You're right. I missed the optional part there. Leave the expression as you wrote it but make sure to add `/login` to the comment above the rule.

---

that is also there 😄

---

Lol. Sorry, must have been stricken with blindness when I did the review 😅

---

happens to the best of us